src/containers/SignIn.js

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react'
+import styled from 'styled-components'
+import { connect } from 'react-redux'
+import { SignInForm, FacebookButton, GoogleButton } from 'twreporter-registration'
+import { setHeaderInfo } from '../actions/header'
+import { AUTHOR_PAGE, LIGHT } from '../constants/index'
+
+const Container = styled.div`
+  margin: 20px 0;
+  display:flex;
+  justify-content: center;
+`
+
+const TITLE = '登入報導者'
+const REDIRECT_PATH = '/'
+
+class SignIn extends Component {
+  componentWillMount() {
+    const { setHeaderInfo } = this.props
+    setHeaderInfo({
+      pageTheme: LIGHT,
+      pageType: AUTHOR_PAGE
+    })
+  }
+
+  render() {
+    const { params } = this.props
+    const redirectPath = params.slug && params.type ? `/${params.type}/${params.slug}` : REDIRECT_PATH
+    return (
+      <Container>
+        <SignInForm
+          title={TITLE}
+          signInRedirectPath={redirectPath}
+          defaultStyle={true}
+          {...this.props}
+        >
+          <FacebookButton />
+          <GoogleButton />
+        </SignInForm>
+      </Container>
+    )
+  }
+}
+
+
+export default connect(null, { setHeaderInfo })(SignIn)
```

Comment:
Isn't [SignIn/SignUp](https://github.com/twreporter/react-redux-registration/blob/b3d885e7a545700240509e3ff3b13aa9bbf78190/src/containers/sign-in-prototype.js#L230) already hooked with redux.connect ?

Reply:
SignUp to a simple component. The SignIn fille need to dispatch setHeaderInfo, so I keep it as a container.